storm-client/src/jvm/org/apache/storm/utils/NimbusClient.java

```diff
@@ -109,12 +109,13 @@ public static NimbusClient getConfiguredClientAs(Map<String, Object> conf, Strin
         }
 
         List<String> seeds;
-        if(conf.containsKey(Config.NIMBUS_HOST)) {
+        if (conf.containsKey(Config.NIMBUS_HOST)) {
             LOG.warn("Using deprecated config {} for backward compatibility. Please update your storm.yaml so it only has config {}",
-                     Config.NIMBUS_HOST, Config.NIMBUS_SEEDS);
+                    Config.NIMBUS_HOST, Config.NIMBUS_SEEDS);
             seeds = Lists.newArrayList(conf.get(Config.NIMBUS_HOST).toString());
         } else {
-            seeds = (List<String>) conf.get(Config.NIMBUS_SEEDS);
+            Object nimbusSeeds = conf.get(Config.NIMBUS_SEEDS);
+            seeds = nimbusSeeds instanceof String[] ? Lists.newArrayList((String[]) nimbusSeeds) : Lists.newArrayList(nimbusSeeds.toString());
```

Comment:
The config is marked as a list of strings.

https://github.com/apache/storm/blob/master/storm-client/src/jvm/org/apache/storm/Config.java#L1049-L1053

But config validation has not run yet.  Perhaps we can validate it before we start using it.